### PR TITLE
Convert raw usage of the session model to the repository

### PR DESCRIPTION
**Pull request by GrahamCampbell**
This is the first part of #1170. I have bigger plans for this though, still to come. Going to end up much faster than it is now. ;)

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Ooooh!

File: app/Config/Repository.php [hunk 1]
```diff
@@ -67,19 +65,25 @@ public function get($name, $default = null)
...
+     * @param string|null $value
      *
      * @return void
      */
     public function set($name, $value)
     {
-        // save the change to the db
-        $this->model->updateOrCreate(compact('name'), compact('value'));
+        if ($value === null) {
+            $this->model->where($name, $value)->delete();
```
Comment (GrahamCampbell) [inline app/Config/Repository.php]: oops - this line is wrong

**Comment by GrahamCampbell** [comment]
@jbrooksuk Any chance you could quickly check this actually works?

**Comment by jbrooksuk** [comment]
I can shortly.

**Comment by GrahamCampbell** [comment]
Ping @jbrooksuk.

**Comment by jbrooksuk** [comment]
Checking now. Anything in particular?

**Comment by jbrooksuk** [comment]
I was able to update settings, so it seems to work!

**Comment by GrahamCampbell** [comment]
Thanks. :)

**Pull request merged**